internal/controller/volsync/vshandler.go

```diff
@@ -1450,9 +1450,17 @@ func (v *VSHandler) getVolumeSnapshotClassFromPVCStorageClass(storageClass *stor
 	var matchedVolumeSnapshotClassName string
 
 	for _, volumeSnapshotClass := range volumeSnapshotClasses {
+		// StorageID From VolumeSnapshotClass should match with storageID in StorageClass if
+		// and only if storageID exists in VolumeSnapshotClass or else skip the check
 		sIDFromSnapClass := volumeSnapshotClass.GetLabels()[StorageIDLabel]
-		if volumeSnapshotClass.Driver == storageClass.Provisioner &&
-			sIDFromSnapClass == storageClass.GetLabels()[StorageIDLabel] {
+		if sIDFromSnapClass != "" {
```

Comment:
I would rewrite this more simply as:
```
if sIDFromSnapClass != "" && sIDFromSnapClass != storageClass.GetLabels()[StorageIDLabel] {
        continue
}
```

We for sure do not need a temp variable to store the match result.

Reply:
done.